actions/notify-slack/action.yml

```diff
@@ -0,0 +1,93 @@
+---
+
+name: 'Notify slack'
+description: 'Posts a slack message with the status of the pipeline'
+
+inputs:
+  application:
+    description: 'The friendly name for the application'
+    type: string
+    required: true
+  data:
+    description: 'The path to the file with data to pass to the render'
+    type: string
+    required: false
+    default: ''
+  emoji:
+    description: 'The github release emoji to display'
+    default: ''
+  environment:
+    description: |
+      The environment variables to pass to the render.  Each environment variable should be in format VAR=VAL with one on each line.
+      Example:
+        environment: |
+          FOO=bar
+          BAR=baz
+    type: string
+    required: false
+    default: ''
+  format:
+    description: 'The format the data file will be in'
+    type: string
+    required: false
+    default: ''
+  version:
+    description: 'The version that is being deployed'
+    type: string
+    default: ${{ github.ref }}
+  type:
+    description: 'The type of action that is being performed such as build or deployment'
+    type: string
+    required: true
+  template:
+    description: 'The path to the template file to render'
+    type: string
+    required: true
+  status:
+    description: 'The status of the deployment.  Options are started, success, and failure'
+    type: string
+    required: true
+  slack-webhook-url:
+    description: 'The slack webhook url to post the message to'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: 'Get info'
+      id: info
+      run: ${{ github.action_path }}/get-info.sh
+      shell: bash
+      env:
```

Comment:
You need to pass in each of the environment variables that the script is expecting.